django/http/multipartparser.py

```diff
@@ -166,15 +171,29 @@ def parse(self):
                 field_name = force_text(field_name, encoding, errors='replace')
 
                 if item_type == FIELD:
+                    # avoid storing more than DATA_UPLOAD_MAX_NUMBER_FIELDS
+                    num_inserted_fields += 1
+                    if settings.DATA_UPLOAD_MAX_NUMBER_FIELDS is not None and settings.DATA_UPLOAD_MAX_NUMBER_FIELDS < num_inserted_fields:
+                        raise SuspiciousOperation('Too many fields')
+
+                    # avoid reading more than DATA_UPLOAD_MAX_MEMORY_SIZE
+                    if settings.DATA_UPLOAD_MAX_MEMORY_SIZE is not None:
+                        read_kwargs = {'size': settings.DATA_UPLOAD_MAX_MEMORY_SIZE - self._data_size}
+                    else:
+                        read_kwargs = {}
```

Comment:
`read_kwargs = {}` before the beginning, and `read_kwargs['size'] = ...` in the loop is both more efficient and more readable IMO.